src/test/java/org/alfresco/repo/event2/CreateRepoEventIT.java

```diff
@@ -0,0 +1,159 @@
+        Thread.sleep(3 * 1000); // 3 seconds time-out
+
+        final RepoEvent<NodeResource> resultRepoEvent = JacksonSerializer.deserialize(futureResult.get(5, SECONDS),RepoEvent.class);
+
+        assertTrue("Repo event retrieved type is "+ resultRepoEvent.getType()+ " .Expected \"Created\"",
+            resultRepoEvent.getType().equals("org.alfresco.event.node.Created"));
+        assertFalse("Repo event ID is not available. ", resultRepoEvent.getId().isEmpty());
+        assertFalse("Repo event source is not available. ", resultRepoEvent.getSource().toString().isEmpty());
+        assertFalse("Repo event subject is not available. ", resultRepoEvent.getSubject().isEmpty());
+        assertFalse("Repo event creation time is not available. ",resultRepoEvent.getTime().toString().isEmpty());
+        assertTrue("Repo event datacontenttype is not application/json. ", resultRepoEvent.getDatacontenttype().equals("application/json"));
+
+        EventData<NodeResource> nodeResourceEventData = resultRepoEvent.getData();
+        assertTrue("Wrong event data principal. ", nodeResourceEventData.getPrincipal().contains("admin"));
+        assertFalse("Event data group ID is not available. ", nodeResourceEventData.getEventGroupId().isEmpty());
+
+        NodeResource nodeResource = nodeResourceEventData.getResource();
+        assertTrue("Resource type is not available. ", nodeResource.getType().toString().contains("NodeResource"));
+
+        assertFalse("Default aspects were not added. ", nodeResource.getAspectNames().isEmpty());
+        assertNull("affectedPropertiesBefore property is not available", nodeResource.getAffectedPropertiesBefore());
+        assertNull("affectedPropertiesAfter property is not available", nodeResource.getAffectedPropertiesAfter());
+        assertNull("aspectNamesBefore property is not available", nodeResource.getAspectNamesBefore());
+        assertNull("aspectNamesAfter property is not available", nodeResource.getAspectNamesAfter());
+
+        String properties = nodeResource.getProperties().toString();
+        assertTrue("Wrong node creator, expecting: admin ", properties.contains("cm:creator=admin"));
+        assertTrue("Wrong node modifier, expecting: admin ", properties.contains("cm:modifier=admin"));
+        assertTrue("Node cm:created property is not available", properties.contains("cm:created"));
+        assertTrue("Node cm:name property is not available", properties.contains("cm:name"));
+        assertTrue("Node cm:modified property is not available", properties.contains("cm:modified"));
+    }
+
+    @Test
+    public void createContentInFolderStructure() throws Exception {
+
+        NodeRef parentRef = createNode(ContentModel.TYPE_CONTAINER);
+        createNode(ContentModel.TYPE_FOLDER,parentRef);
+        Thread.sleep(3 * 1000); // 3 seconds time-out
+
+        final RepoEvent resultRepoEvent = JacksonSerializer.deserialize(futureResult.get(5, SECONDS),RepoEvent.class);
```

Comment:
Same here - use acs-event-model ObjectMapper. I'm not going to add a comment everywhere that you deserialized the JSON, so remember to change them all.